nimble/host/src/ble_store_util.c

```diff
@@ -230,13 +299,35 @@ ble_store_util_delete_oldest_peer(void)
 int
 ble_store_util_status_rr(struct ble_store_status_event *event, void *arg)
 {
+    int rc = BLE_HS_EUNKNOWN;
+    ble_addr_t peer_id_addr;
+    int num_peers;
+
     switch (event->event_code) {
     case BLE_STORE_EVENT_OVERFLOW:
         switch (event->overflow.obj_type) {
             case BLE_STORE_OBJ_TYPE_OUR_SEC:
             case BLE_STORE_OBJ_TYPE_PEER_SEC:
             case BLE_STORE_OBJ_TYPE_CCCD:
-                return ble_gap_unpair_oldest_peer();
```

Comment:
I believe that for CCCD we should have different handling and if we want to unpair older peer we should not unpair peer for which we store CCCD.  If there is nothing to unpair just return the error.

In the CCCD event there is and ble_addr we should use to make sure that one is no unpaired.

Reply:
@rymanluk agreed !! I have made relevant changes. For CCCD,  if `ble_gap_unpair_oldest_peer` is unable to delete any peer, we continue to find other peers occupying CCCDs in storage. I have also modified callback function (`ble_store_util_iter_peer_cccd`) to skip the current (i.e. for which CCCDs need to be stored) peer entry.